File: backend/app/Models/Appliance.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class Appliance extends Model
{
    use HasFactory, HasUuids;

    public $incrementing = false;
    protected $keyType = 'string';

    protected $fillable = [
        'name',
        'category',
        'power_watts',
        'daily_use_hours',
        'is_on',
    ];

    protected $casts = [
        'is_on' => 'boolean',
        'daily_use_hours' => 'float',
    ];

    public function usageLogs(): HasMany
    {
        return $this->hasMany(UsageLog::class);
    }

    public function getDailyKwhAttribute(): float
    {
        return ($this->power_watts / 1000) * $this->daily_use_hours;
    }
}

File: frontend/lib/features/dashboard/domain/entities/appliance.dart
import '../services/consumption_calculator.dart';

class Appliance {
  const Appliance({
    required this.id,
    required this.name,
    required this.category,
    required this.powerW,
    required this.dailyUseHours,
    required this.isOn,
  });

  final String id;
  final String name;
  final String category;
  final double powerW;
  final double dailyUseHours;
  final bool isOn;

  double get dailyKwh => scheduledDailyKwh(
    powerWatts: powerW,
    dailyUseHours: dailyUseHours,
    isOn: isOn,
  );

  Appliance copyWith({
    String? id,
    String? name,
    String? category,
    double? powerW,
    double? dailyUseHours,
    bool? isOn,
  }) {
    return Appliance(
      id: id ?? this.id,
      name: name ?? this.name,
      category: category ?? this.category,
      powerW: powerW ?? this.powerW,
      dailyUseHours: dailyUseHours ?? this.dailyUseHours,
      isOn: isOn ?? this.isOn,
    );
  }
}
